src/couch_external_server.erl

```diff
@@ -40,47 +38,52 @@ execute(Pid, JsonReq) ->
 init([Name, Command]) ->
     couch_log:info("EXTERNAL: Starting process for: ~s", [Name]),
     couch_log:info("COMMAND: ~s", [Command]),
+    ok = config:subscribe_for_changes([{"couchdb", "os_process_timeout"}]),
     process_flag(trap_exit, true),
     Timeout = list_to_integer(config:get("couchdb", "os_process_timeout",
         "5000")),
     {ok, Pid} = couch_os_process:start_link(Command, [{timeout, Timeout}]),
-    ok = config:listen_for_changes(?MODULE, Pid),
-    {ok, {Name, Command, Pid}}.
+    {ok, {Name, Command, Pid, whereis(config_event)}}.
```

Comment:
Seems like a lot of extra work just for that one 'EXIT' clause. I do see that without it we'd enter that clause for any 'EXIT' rather than throwing a function clause error. Though given that its rare I'd probably just undo the state tuple change and if we get an unknown 'EXIT' message we just erlang:error on it.